drivers/include/dht.h

```diff
@@ -56,11 +59,24 @@ typedef struct {
     uint16_t temperature;   /**< temperature in deci-Celsius */
 } dht_data_t;
 
+/**
+ * @brief   Device type of the DHT device
+ */
+typedef enum {
+    DHT11,                  /**< DHT11 device identifier */
+    DHT22,                  /**< DHT22 device identifier */
+    DHT21 = DHT22,          /**< DHT21 device identifier */
+    AM2301 = DHT22,         /**< AM2301 device identifier */
+} dht_type_t;
```

Comment:
```suggestion
typedef enum {
    DHT11,                  /**< Older DHT11 variants with either 1 °C or
                              *  0.1 °C resolution */
    DHT11_2022,             /**< New DHT11 variant with 0.01 °C resolution */
    DHT22,                  /**< DHT22 */
    DHT21 = DHT22,          /**< DHT21 */
    AM2301 = DHT22,         /**< AM2301 */
} dht_type_t;
```

How about this instead? (Not insisting on the naming here.) The `lsb_is_hundreds` would only make sense for DHT11 device and would be wasted for e.g. DTH2x device.